apis/scheduling/v1alpha1/reservation_types.go

```diff
@@ -62,8 +62,29 @@ type ReservationSpec struct {
 	// +kubebuilder:default=true
 	// +optional
 	AllocateOnce *bool `json:"allocateOnce,omitempty"`
+	// AllocatePolicy represents the allocation policy of reserved resources that Reservation expects.
+	// +kubebuilder:validation:Enum=Aligned;Restricted
+	// +optional
+	AllocatePolicy ReservationAllocatePolicy `json:"allocatePolicy,omitempty"`
 }
 
+type ReservationAllocatePolicy string
+
+const (
+	// ReservationAllocatePolicyDefault means that there is no restriction on the policy of reserved resources,
+	// and allocated from the Reservation first, and if it is insufficient, it is allocated from the node.
+	ReservationAllocatePolicyDefault ReservationAllocatePolicy = ""
+	// ReservationAllocatePolicyAligned indicates that the Pod allocates resources from the Reservation first.
+	// If the remaining resources of the Reservation are insufficient, it can be allocated from the node,
+	// but it is required to strictly follow the resource specifications of the Pod.
+	// This can be used to avoid the problem that a Pod uses multiple Reservations at the same time.
+	ReservationAllocatePolicyAligned ReservationAllocatePolicy = "Aligned"
+	// ReservationAllocatePolicyRestricted means that the resources requested by the Pod overlap with the resources reserved by the Reservation,
+	// then these intersection resources can only be allocated from the Reservation, but other resources can be allocated from the node.
```

Comment:
> but other resources can be allocated from the node

How to understand the `other resources`, is the resource claimed by the Pod but not by the Reservation?

Reply:
Other resources refer to resources other than the intersection of Pod and Reservation. For example, if a Pod applies for 4C8Gi and 1 ENI, and the Reservation reserves 4C8Gi, then 1 ENI is other resource.

Reply:
Add a comment `(resources declared in the Pod but not reserved in the Reservation)` to clarify the semantics?

Reply:
Good suggestion, I will update it.

Reply:
updated.